src/fidesops/ops/service/connectors/saas/authenticated_client.py

```diff
@@ -75,27 +81,77 @@ def get_authenticated_request(
         # otherwise just return the prepared request
         return req
 
+    def retry_send(  # type: ignore
+        retry_count: int,
+        backoff_factor: float,
+        retry_status_codes: List[int] = [429, 502, 503, 504],
+    ) -> Callable:
+        """
+        Retry decorator for http requests, backing off exponentially or listening to server retry-after header
+
+        Exponential backoff factor uses the following formula:
+        backoff_factor * (2 ** (retry_attempt))
+        For an backoff_factor of 1 it will sleep for 2,4,8 seconds
+
+        General exceptions are always retried. RequestFailureResponseException exceptions are only retried
```

Comment:
this struck me as a bit counter-intuitive. but i suppose we're just sort of "playing it safe" here, and assuming we should retry if we don't know more about the exception? i'd just like to understand the reasoning a bit better on why we always retry on "general" exceptions (which, to me, seem more likely to be caused by bad config or some user error) but are more discerning of the `RequestFailureResponseException`s

Reply:
Also discussed this offline. We decided to change this so that general exceptions are not retries. I also added a comment around this.